src/PFire.Core/Protocol/Messages/Inbound/LoginRequest.cs

```diff
@@ -21,15 +21,16 @@ public override void Process(IXFireClient context)
             var user = context.Server.Database.QueryUser(Username);
             if (user != null)
             {
-                if (user.Password != Password)
+                if (!BCrypt.Net.BCrypt.Verify(Password, user.Password))
```

Comment:
Sorry I'm still not following the purpose of this. The password is already hashed on the client side (the salt is forwarded to the client on login, which is eh), so at this point right now, the password is currently hashed and not plain text.

Wouldn't an attacker just be able to MITM the hashed password, rehash it using this same library and be able to login?

Reply:
The server sends a salt to the client as part of the loginChallege message

Reply:
Idea: store salt in the database, so it's not stored in code, and then use that to re-hash the password we get from the client.

Next Login, load the user's rehased password, hash the one they sent, see if they are the same.

The salt can be generated for that user, when they first login (i.e the signup login).

Reply:
@darcymiranda The purpose on this is that we cannot reuse the password that exists in the database. If I replace the client with another application, then this application wouldn't hash and send it as plaintext. The server side treat it always as plaintext. It doesn't matter that the client side hashed it already. 

Of course, this application isn't at high risk because we are not saving very valuable information. But it is a good practice that we always hashing one time at server side. This packages saves the salt nearby the password (Same field). Every password use a different salt. That is why it looks like that we don't save the salt, but it does. We use this package because this is a proven way. 

You are right that if somebody captured someone his password within a MITM attack, then this security change doesn't help. But we reduce other risks with this change.

Reply:
Ok I agree with some of your points. If anything, this doesn't present any harm that I can tell

Reply:
Oops I introduced a conflict when merging in other changes. Can you please fix that up? I'll merge first priority after

Reply:
@darcymiranda I fixed the merge.